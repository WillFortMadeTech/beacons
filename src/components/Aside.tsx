import React, { FunctionComponent, ReactNode } from "react";
import styles from "../styles/Aside.module.scss";

interface AsideProps {
  title: string;
  children: ReactNode;
}

const Aside: FunctionComponent<AsideProps> = ({
  title,
  children,
}: AsideProps): JSX.Element => (
  <>
    <aside className={styles.related_items} role="complementary">
      <h2 className="govuk-heading-m" id="subsection-title">
        {title}
      </h2>
      <nav role="navigation" aria-labelledby="subsection-title">
        {children}
      </nav>
    </aside>
  </>
);

export default Aside;
