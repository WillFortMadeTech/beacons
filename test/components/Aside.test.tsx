import React from "react";
import { render } from "@testing-library/react";
import Aside from "../../src/components/Aside";

describe("Aside", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <Aside title={"Beacons registration"}>
        <MyComponent />
      </Aside>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

const MyComponent = () => <h1>Beacons Registration</h1>;
