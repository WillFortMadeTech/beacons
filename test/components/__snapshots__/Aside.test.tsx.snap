// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside renders correctly 1`] = `
<DocumentFragment>
  <aside
    class="related_items"
    role="complementary"
  >
    <h2
      class="govuk-heading-m"
      id="subsection-title"
    >
      Beacons registration
    </h2>
    <nav
      aria-labelledby="subsection-title"
      role="navigation"
    >
      <h1>
        Beacons Registration
      </h1>
    </nav>
  </aside>
</DocumentFragment>
`;
